### DISCUSSION TITLE
Clean Japanese text feature?

### DISCUSSION THREAD

**Discussion by HubKing** [VoxelCubes/PanelCleaner#97] [Ideas]
Maybe, this doesn't fit the purpose of this app, but how about adding this feature, or creating a spin-off app that can do this? A lot of scanned images have blurry Japanese text, especially the Chinese characters. Since this app can already detect the Japanese text's position, how about instead of removing the Japanese text, enhance the text with high-resolution font (up-scale the image and replace the text with higher-resolution)?

Most Japanese texts are typefaces, not handwriting, so it probably should not be very difficult to A.I. to read it. It's just I don't know anything about A.I., so I can't create an app like that myself. Something like below:

original

![image](https://github.com/VoxelCubes/PanelCleaner/assets/10900489/2bab1f4d-8396-4c65-9b89-49e1ff63eb4f)

enhanced

![image](https://github.com/VoxelCubes/PanelCleaner/assets/10900489/9a0dbf11-f773-40e4-9dd6-06f1181da260)

**Comment by VoxelCubes** [comment]
Man, that's a difficult problem, since the blurrier it is, the harder it gets to reliably OCR the characters. And then enhancing them without having a crude background like in your mockup (otherwise, isn't that jusy a bit too ugly? Or what woukd this be used for?) Isn't so straightforward with non-digital art. You'd basically need an inpainter that can reliably draw text, or something that reliably places text where it belongs.

That isn't something that's possible yet, definitely not with the AI models in panel cleaner, but in the future, large language models are getting incredibly good at this, see things like GPT4o. I mighty be possible to build this when such models are available for offline use, it'd be amazing for sure.

But for now, I don't think it's feasible. Interesting idea though, I'll consider it in the future.

**Reply by HubKing** [reply]
Thanks for the reply. With your reply, I now get that this is a hard problem for A.I. and I can give it up. Many scans were blurry but due to low resolution, but there rarely were letters that were unrecognisable to me (i.e., I could all read them, but they looked ugly and made my eyes tired), so after hearing all these A.I.-hypes these days, I thought maybe A.I. could do it now, but I guess A.I.'s are still not even smart as an ordinary, not-smart human being, for whom this kind of task is trivial.

Come to think of it, recently I asked free A.I.'s (Google Gemini, Copilot, and ChatGpt 4o) to recognise a big screen capture of "赫鱗"  

![Screenshot_20240512_222526](https://github.com/VoxelCubes/PanelCleaner/assets/10900489/802d7cca-58b0-40d2-beec-e55ac2986d82)

and none of them correctly recognised the Chinese letters. Any Japanese or Chinese person would have at least got the shapes correctly, even if they don't know the letters, but those A.I.'s got it either completely wrong or made mistakes. The closest reply was "赫麟".

But who knows, maybe in 10 years, A.I. could really become as smart as an ordinary human. I wish I could live up to see that.

**Reply by VoxelCubes** [reply]
It's the type of problem that AI can do, but not well yet. Gonna have to wait a while longer. It's pattern matching.

But don't forget one important fact: the i in LLM stands for intelligence. That is to say, it doesn't exist, there is no intelligence in large language models, which are what this "AI" of current day typically is. They are systems that, given what it has seen so far, try to predict the next letter to print to your screen. There is no rational thought process here, just probabilities. So don't expect too much.

**Reply by HubKing** [reply]
> the i in LLM stands for intelligence. That is to say, it doesn't exist, there is no intelligence in large language models

Gonna use that in the next job interview, if they ask about A.I., which they probably will...

**Reply by VoxelCubes** [reply]
Here is an article about people using LLMs to submit bogus security issues to projects, wasting everyone's time. It isn't the first one to use the phrase, but boy is it ever applicable here: https://daniel.haxx.se/blog/2024/01/02/the-i-in-llm-stands-for-intelligence/
Machine Learning can do pattern matching well, but especially for complex things like security research, it's abysmal.
